src/ast/nodes/LogicalExpression.ts

```diff
@@ -103,6 +103,7 @@ export default class LogicalExpression extends NodeBase implements Deoptimizable
 		recursionTracker: PathTracker,
 		origin: DeoptimizableEntity
 	): LiteralValueOrUnknown {
+		if (origin === this) return UnknownValue;
```

Comment:
Actually a good idea. Alternatively, one could have used the recursionTracker, but this is of course also correct. Makes me wonder, though, if there are more situations where this could happen. But considering the bug at hand, this looks good to me.

Reply:
> Makes me wonder, though, if there are more situations where this could happen, i.e. where handling "getLiteralValue" might profit from an `origin` check.

Yeah, but it might be hard to find those situations without issue reports.